Lib/test/test_sqlite3/test_transactions.py

```diff
@@ -327,5 +328,103 @@ def test_isolation_level_none(self):
         self.assertEqual(self.traced, [self.QUERY])
 
 
+class AutocommitPEP249(unittest.TestCase):
+    """Test PEP-249 compliant autocommit behaviour."""
+    @contextmanager
+    def check_stmt_trace(self, cx, expected):
+        try:
+            traced = []
+            cx.set_trace_callback(lambda stmt: traced.append(stmt))
+            yield
+        finally:
+            self.assertEqual(traced, expected)
+            cx.set_trace_callback(None)
+
+    def test_autocommit_default(self):
+        with memory_database() as cx:
+            self.assertEqual(cx.autocommit,
+                             sqlite.DEPRECATED_TRANSACTION_CONTROL)
+
+    def test_autocommit_setget(self):
+        dataset = (
+            True,
+            False,
+            sqlite.DEPRECATED_TRANSACTION_CONTROL,
+        )
+        for mode in dataset:
+            with self.subTest(mode=mode):
+                with memory_database(autocommit=mode) as cx:
+                    self.assertEqual(cx.autocommit, mode)
+                with memory_database() as cx:
+                    cx.autocommit = mode
+                    self.assertEqual(cx.autocommit, mode)
+
+    def test_autocommit_setget_invalid(self):
+        msg = "autocommit must be True, False, or.*DEPRECATED"
+        for mode in "a", 12, (), None:
+            with self.subTest(mode=mode):
+                with self.assertRaisesRegex(ValueError, msg):
+                    with memory_database(autocommit=mode) as cx:
+                        self.assertEqual(cx.autocommit, mode)
+
+    def test_autocommit_disabled(self):
+        expected = [
+            "SELECT 1",
+            "COMMIT",
+            "BEGIN",
+            "ROLLBACK",
+            "BEGIN",
+        ]
+        with memory_database(autocommit=False) as cx:
+            self.assertTrue(cx.in_transaction)
```

Comment:
For the reviewers convenience: the `in_transaction` property is _directly coupled_ with the return result of [sqlite3_get_autocommit](https://sqlite.org/c3ref/get_autocommit.html):

https://github.com/python/cpython/blob/7e9eaad864349d2cfd4c9ffc4453aba03b2cbc16/Modules/_sqlite/connection.c#L1589-L1598